x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -645,27 +650,81 @@ private static void checkFilterMatchConditions(LogicalPlan plan, Set<Failure> fa
-                                lp.sourceText().split(" ")[0].toUpperCase(Locale.ROOT)
-                            )
-                        );
-                    }
-                });
-            }
+            checkCommandsAfterQueryStringFunction(plan, condition, failures);
+            checkFullTextFunctionsConditions(condition, failures);
+            checkFullTextFunctionsParents(condition, failures);
         } else {
             plan.forEachExpression(FullTextFunction.class, ftf -> {
                 failures.add(fail(ftf, "[{}] function is only supported in WHERE commands", ftf.functionName()));
             });
         }
     }
+
+    private static void checkCommandsAfterQueryStringFunction(LogicalPlan plan, Expression condition, Set<Failure> failures) {
+        condition.forEachDown(QueryStringFunction.class, qsf -> {
+            plan.forEachDown(LogicalPlan.class, lp -> {
+                if ((lp instanceof Filter || lp instanceof OrderBy || lp instanceof EsRelation) == false) {
+                    failures.add(
+                        fail(
+                            plan,
+                            "[{}] function cannot be used after {}",
+                            qsf.functionName(),
+                            lp.sourceText().split(" ")[0].toUpperCase(Locale.ROOT)
+                        )
+                    );
+                }
+            });
+        });
+    }
+
+    private static void checkFullTextFunctionsConditions(Expression condition, Set<Failure> failures) {
+        condition.forEachUp(Or.class, or -> {
+            Expression left = or.left();
+            Expression right = or.right();
+            left.forEachDown(FullTextFunction.class, ftf -> {
+                if (canPushToSource(right, x -> false) == false) {
+                    failures.add(
+                        fail(
+                            or,
+                            "Invalid condition [{}]. Function {} can't be used as part of an or condition that includes [{}]",
+                            or.sourceText(),
+                            ftf.functionName(),
+                            right.sourceText()
+                        )
+                    );
+                }
+            });
+            right.forEachDown(FullTextFunction.class, ftf -> {
+                if (canPushToSource(left, x -> false) == false) {
+                    failures.add(
+                        fail(
+                            or,
+                            "Invalid condition [{}]. Function {} can't be used as part of an or condition that includes [{}]",
+                            or.sourceText(),
+                            ftf.functionName(),
+                            left.sourceText()
+                        )
+                    );
+                }
+            });
+        });
+    }
+
+    private static void checkFullTextFunctionsParents(Expression condition, Set<Failure> failures) {
```

Comment:
This check was added so the following is not allowed - would otherwise cause a downstream error:

`from test | where match(first_name, "Anna") in ("hello", "world")`

`from test | where match(first_name, "Anna") is null`